@xen-orchestra/backups/disks/RemoteDisk.mjs

```diff
@@ -0,0 +1,81 @@
+import { RandomAccessDisk } from "@xen-orchestra/disk-transform"
+
+/**
+ * @typedef {import('@xen-orchestra/disk-transform').DiskBlock} DiskBlock
+ * @typedef {import('./RemoteDisk.mjs').DiskMetadata} DiskMetadata
+ * 
+ * @typedef {(Object)} DiskMetadata
+ * @property {(string)} uuid
+ * @property {(string)} parentId
+ * @property {(number)} virtualSize
+ * @property {(string)} parentLocation
+ */
+
+
+export class RemoteDisk extends RandomAccessDisk {
+    /**
+     * Abstract
+     * @returns {string}
+     */
+    getPath() { }
+
+    /**
+     * Abstract
+     * @returns {string}
+     */
+    getUuid() { }
+
+    /**
+     * Abstract
+     * @returns {number}
+     */
+    getMaxTableEntries() { }
```

Comment:
I think we do not need to have a separate method getMaxTableEntries since Disk already expose getVirtualSize() and getBlockSize()

If you need this method, it can be added directly to the Disk class, with a more generic name, like `getMaxBlockCount` or `getAddressableBlocks`